src/main/java/org/java_lcw/jil/JavaImage.java

```diff
@@ -0,0 +1,757 @@
+package org.java_lcw.jil;
+
+import java.awt.image.BufferedImage;
+import java.awt.image.DataBufferByte;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayDeque;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Random;
+
+import org.java_lcw.jil.scalers.BiCubicScaler;
+import org.java_lcw.jil.scalers.BiLinearScaler;
+import org.java_lcw.jil.scalers.NearestNeighborScaler;
+
+
+/**
+ *
+ * Main Image object used to construct new Image files.  
+ * 
+ * @author lcw - Luke Wahlmeier
+ */
+public class JavaImage implements Image {
```

Comment:
A possible better name than JavaImage would be "JilImage" (though typing it, the l and capital I looks weird next to each other).

Technically AWT is a "java" Image as well.  I feel like "JilImage" better describes what it is.